Menus/MainMenuGUI.cpp

```diff
@@ -1,3263 +1,539 @@
-
-				m_KnownMods.push_back(r);
-			}
-		}
-    }
-
-	// Now add missing data from disabled mods settings
-	map<string, bool> disabledMods = g_SettingsMan.GetDisabledModsList();
-	for(map<std::string, bool>::iterator itr = disabledMods.begin(); itr != disabledMods.end(); ++itr)
-	{
-		string modPath = itr->first;
-		std::transform(modPath.begin(), modPath.end(), modPath.begin(), ::tolower);
-
-		bool found = false;
-
-		// Check if this mod is already in list
-		for (vector<ModRecord>::iterator mItr = m_KnownMods.begin(); mItr != m_KnownMods.end(); ++mItr)
-			if (modPath == (*mItr).ModulePath)
-				found = true;
-
-		if (!found)
-		{
-			ModRecord r;
-			r.ModulePath = modPath;
-			r.Description = "n/a, module not loaded";
-			r.ModuleName = "n/a, module not loaded";
-			r.Disabled = itr->second;
-
-			m_KnownMods.push_back(r);
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void MainMenuGUI::CreateMainScreen() {
+		m_MainMenuScreens.at(MenuScreen::MainScreen) = dynamic_cast<GUICollectionBox *>(m_MainMenuScreenGUIControlManager->GetControl("MainScreen"));
+		m_MainMenuScreens.at(MenuScreen::MainScreen)->CenterInParent(true, false);
+
+		m_MainMenuButtons.at(MenuButton::CampaignButton) = dynamic_cast<GUIButton *>(m_MainMenuScreenGUIControlManager->GetControl("ButtonMainToCampaign"));
+		m_MainMenuButtons.at(MenuButton::CampaignButton)->SetText("MetaGame");
+		m_MainMenuButtons.at(MenuButton::ScenarioButton) = dynamic_cast<GUIButton *>(m_MainMenuScreenGUIControlManager->GetControl("ButtonMainToSkirmish"));
+		m_MainMenuButtons.at(MenuButton::MultiplayerButton) = dynamic_cast<GUIButton *>(m_MainMenuScreenGUIControlManager->GetControl("ButtonMainToMultiplayer"));
+		m_MainMenuButtons.at(MenuButton::SettingsButton) = dynamic_cast<GUIButton *>(m_MainMenuScreenGUIControlManager->GetControl("ButtonMainToOptions"));
+		m_MainMenuButtons.at(MenuButton::ModManagerButton) = dynamic_cast<GUIButton *>(m_MainMenuScreenGUIControlManager->GetControl("ButtonMainToModManager"));
+		m_MainMenuButtons.at(MenuButton::EditorsButton) = dynamic_cast<GUIButton *>(m_MainMenuScreenGUIControlManager->GetControl("ButtonMainToEditor"));
+		m_MainMenuButtons.at(MenuButton::CreditsButton) = dynamic_cast<GUIButton *>(m_MainMenuScreenGUIControlManager->GetControl("ButtonMainToCreds"));
+		m_MainMenuButtons.at(MenuButton::QuitButton) = dynamic_cast<GUIButton *>(m_MainMenuScreenGUIControlManager->GetControl("ButtonQuit"));
+		m_MainMenuButtons.at(MenuButton::ResumeButton) = dynamic_cast<GUIButton *>(m_MainMenuScreenGUIControlManager->GetControl("ButtonResume"));
+
+		for (int mainScreenButton = 0; mainScreenButton < 9; ++mainScreenButton) {
+			m_MainMenuButtons.at(mainScreenButton)->CenterInParent(true, false);
+			std::string buttonText = m_MainMenuButtons.at(mainScreenButton)->GetText();
+			std::transform(buttonText.begin(), buttonText.end(), buttonText.begin(), ::toupper);
```

Comment:
This was more of an issue when we had `using namespace std;` because it would try to use the `std` ones instead if it wasn't qualified.

I don't know if we should keep the global namespace qualifier tho, cause we don't use it whenever we're calling other C functions, but it does make it clear that this isn't using the `std` ones.

And the reason i'm not using the `std` ones is because having to do a lambda to use it is trash. So i'm not changing it.

Reply:
Hmmm, I'm on the fence about specifying global namespace here. As you say, we don't do it in other places so it's a bit weird. But on the other hand, I can definitely see someone thinking it's std but missing the std qualifier.
Though as you say, I guess in order to use std functions here we'd need the std qualifier anyway, so maybe it's fine to delete it? I'm not sure, this one's a bit tricky.

Reply:
I'm just gonna keep it as it is and continue to use the qualifier for these in future:tm: uses of `std::transform`.